images/virtualization-artifact/pkg/controller/vd/internal/resizing.go

```diff
@@ -85,33 +86,58 @@ func (h ResizingHandler) Handle(ctx context.Context, vd *virtv2.VirtualDisk) (re
 		return reconcile.Result{}, errors.New("pvc not found for ready virtual disk")
 	}
 
-	if newSize.Equal(pvc.Status.Capacity[corev1.ResourceStorage]) {
-		if condition.Reason == vdcondition.InProgress {
-			condition.Status = metav1.ConditionTrue
-			condition.Reason = vdcondition.Resized
-			condition.Message = ""
-			return reconcile.Result{}, nil
-		}
-
+	pvcSpecSize := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+	switch vdSpecSize.Cmp(pvcSpecSize) {
+	// Expected disk size is LESS THAN expected pvc size: no resize needed as resizing to a smaller size is not possible.
+	case -1:
 		condition.Status = metav1.ConditionFalse
 		condition.Reason = vdcondition.NotRequested
 		condition.Message = ""
```

Comment:
I think we should say something to user about "LESS" situation, as it was with ErrTooSmallDiskSize error.

```suggestion
		condition.Message = fmt.Sprintf("The virtual disk size is too low. Should be >= %s", pvcSpecSize.String())
```